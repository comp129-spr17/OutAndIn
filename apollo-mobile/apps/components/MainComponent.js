'use strict'

import React, { Component } from 'react';
import { StyleSheet, AppRegistry, Text, ListView, TextInput, TouchableOpacity, View, ScrollView } from 'react-native';

const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 != r2});
var dataArray=[];
class MainComponent extends Component
{
  constructor() {
    super();
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 != r2});
    this.state = {
      message: '',
      dataSource: ds.cloneWithRows(dataArray)
    };
  }

  render() {
    return (
      <ScrollView style={styles.container}>
        <View style={styles.header}>
          <Text style={styles.Logo}> Apollo </Text>
        </View>
        <ListView dataSource = {this.state.dataSource}
                  enableEmptySections = {true}
                  style = {styles.listMessage}
                  renderRow = {
                    (rowData) =>
                      <View style={styles.message}>
                        <Text>{rowData}</Text>
                      </View>
                  }
        />
        <View style={styles.containerInput}>
          <TextInput style={styles.inputMessage}
                     onChangeText={text => this.setState({message: text})}
                     value={this.state.message}
          />
          <TouchableOpacity style={styles.buttonSend}>
            <Text style = {styles.send}> SEND </Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    )
  }
}

const styles  = StyleSheet.create({
  send:
  {
    color: 'white',
    fontWeight: 'bold',
    //fontFamily: 'FontAwesome'
  },
  header:
  {
    height:70,
    //flexDirection: "column", /*top to bottom*/
    //justifyContent: "flex-start", /*start at the top*/
    //alignItems: "stretch", /*all components take up full width*/
    backgroundColor: '#F8F8F8'
  },
  Logo:
  {
    textAlign: 'center',
    color: '#424242',
    fontSize: 16,
    marginTop: 30
  },
  container:
  {
    //marginTop: 20,
    flex: 1,
    flexDirection: 'column'
  },
  listMessage:
  {
    flex: 13,
    paddingLeft: 5,
    paddingRight:5
  },
  message:
  {
    paddingTop: 5,
    paddingRight:5,
    paddingLeft:5
  },
  containerInput: {
    height: 40,
    flexDirection: 'row',
  },
  inputMessage: {
    flex: 8,
    borderWidth: 1,
    borderColor: '#164762',
    paddingLeft: 5
  },
  buttonSend: {
    flex: 2,
    backgroundColor: '#164762',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
  }
})

module.exports = MainComponent
