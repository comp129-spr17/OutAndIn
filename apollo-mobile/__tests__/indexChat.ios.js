import 'react-native';
import React from 'react';
import MainComponent from '../apps/components/MainComponent';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <MainComponent />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
