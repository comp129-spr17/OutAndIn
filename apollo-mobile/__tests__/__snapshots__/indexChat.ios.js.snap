exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <View>
    <RCTScrollView>
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "height": 80,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "row",
          "height": 40,
        }
      }>
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "borderColor": "#EDEDED",
            "borderWidth": 1,
            "flex": 8,
            "paddingLeft": 5,
          }
        }
        value="" />
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "row",
          "height": 40,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "color": "#8A8A8A",
              "paddingLeft": 7,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "color": "#8A8A8A",
              "paddingLeft": 7,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "color": "#8A8A8A",
              "paddingLeft": 7,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "color": "#8A8A8A",
              "paddingLeft": 7,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
  </View>
</View>
`;
